storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -137,15 +138,8 @@ public RemoteIndexCache(long maxSize, RemoteStorageManager remoteStorageManager,
     public void resizeCacheSize(long remoteLogIndexFileCacheSize) {
         lock.writeLock().lock();
         try {
-            // When resizing the cache, we always start with an empty cache. There are two main reasons:
-            // 1. Resizing the cache is not a high-frequency operation, and there is no need to fill the data in the old
-            // cache to the new cache in time when resizing inside.
-            // 2. Since the eviction of the caffeine cache is cleared asynchronously, it is possible that after the entry
-            // in the old cache is filled in the new cache, the old cache will clear the entry, and the data in the two caches
-            // will be inconsistent.
-            internalCache.invalidateAll();
-            log.info("Invalidated all entries in the cache and triggered the cleaning of all index files in the cache dir.");
-            internalCache = initEmptyCache(remoteLogIndexFileCacheSize);
+            internalCache.policy().eviction().orElseThrow(() -> new NoSuchElementException("No eviction policy is set for the remote index cache.")
```

Comment:
Why aren't we changing this in initEmptyCache() itself? Right now, setting the eviction policy is done separately in two places, in `resizeCacheSize` and in `initEmptyCache`. Alternatively, we can call `initEmptyCache` from both these places. It's just that in the case of `resizeCacheSize`, we will reset the maxSize of the cache to the new value.

Reply:
Ah! I think the difference is that we use a builder to construct a new cache in initCache() whereas we want to access a method (policy()) on existing cache over here.